pkg/registry/core/pod/storage/eviction.go

```diff
@@ -220,7 +221,7 @@ func (r *EvictionREST) Create(ctx context.Context, name string, obj runtime.Obje
 		pdbName = pdb.Name
 
 		// If the pod is not ready, it doesn't count towards healthy and we should not decrement
-		if !podutil.IsPodReady(pod) && pdb.Status.CurrentHealthy >= pdb.Status.DesiredHealthy && pdb.Status.DesiredHealthy > 0 {
+		if !podutil.IsPodReady(pod) && enoughHealthyReplicas(pdb) {
```

Comment:
If the pod is ready=F, why would we ever want to count it against disruption?  IOW, why isn't this check simply 

```go
if !podutil.IsPodReady(pod) {
  requireCurrentPodResourceVersionForDeletion=true
  return nil
}
```

Reply:
Other pods might be disrupted, that's the whole point of checking the PDB.  If we return nil here as you suggest, it's basically saying "we can delete this pod, PDB says we're good."

We only want to return nil here if the pod is not ready (eg, we're draining a failed node) and there are enough healthy replicas elsewhere (checking the PDB) AND (new in this patch) the PDB doesn't have a status indicating it was unable to accurately get pod counts.

Reply:
If a pod is not ready, Why isn't it always valid to delete it regardless of how many other present?

Reply:
Because that's not how the API behavior is defined.

Reply:
> Because that's not how the API behavior is defined.

Can I get a reason that explains and defends the choice of that API behavior in the code comment (and here if you like) so that a future reviewer can understand why its good to do this?  It doesn't seem like an obvious choice to me, since evicting a not-ready pod seems strictly better than most other choices.